src/test/java/com/pnu/dev/shpaltaif/integration/service/PublicAccountServiceIntegrationTest.java

```diff
@@ -8,183 +8,164 @@
+                .description(publicAccountDto.getDescription())
+                .user(this.userWriter)
                 .build();
 
-        PublicAccount actualPublicAccount = publicAccountsAfterCreate.get(0);
-        assertThat(actualPublicAccount)
+        assertThat(createdPublicAccount)
                 .isEqualToIgnoringGivenFields(expectedPublicAccount, "id", "createdAt", "updatedAt");
 
         // Read
-        PublicAccount foundByIdPublicAccount = publicAccountService.findById(publicAccountsAfterCreate.get(0).getId());
-        assertEquals(actualPublicAccount, foundByIdPublicAccount);
+        PublicAccount foundByIdPublicAccount = publicAccountService.findById(createdPublicAccount.getId());
+        assertEquals(createdPublicAccount, foundByIdPublicAccount);
 
         // Update
         PublicAccountDto updatePublicAccountDto = PublicAccountDto.builder()
-                .name(UPDATED_NAME)
-                .surname(UPDATED_SURNAME)
-                .profileImageUrl(UPDATED_PROFILE_IMAGE_URL)
-                .description(UPDATED_DESCRIPTION)
+                .name("updated_name")
+                .surname("updated_surname")
+                .profileImageUrl("updated_profile_image_url")
+                .description("updated_description")
                 .build();
 
-        publicAccountService.update(updatePublicAccountDto, actualPublicAccount.getId());
+        publicAccountService.update(updatePublicAccountDto, createdPublicAccount.getId());
 
-        List<PublicAccount> publicAccountsAfterUpdate = publicAccountService.findAll();
-        assertEquals(1, publicAccountsAfterUpdate.size());
+        //Verify whether the number of PublicAccounts in database hasn't changed after update
+        assertEquals(expectedPublicAccountsNumber, publicAccountService.findAll().size());
 
         PublicAccount expectedUpdatedPublicAccount = PublicAccount.builder()
-                .name(UPDATED_NAME)
-                .surname(UPDATED_SURNAME)
-                .profileImageUrl(UPDATED_PROFILE_IMAGE_URL)
-                .description(UPDATED_DESCRIPTION)
-                .user(user)
+                .name(updatePublicAccountDto.getName())
+                .surname(updatePublicAccountDto.getSurname())
+                .profileImageUrl(updatePublicAccountDto.getProfileImageUrl())
+                .description(updatePublicAccountDto.getDescription())
+                .user(this.userWriter)
                 .build();
 
-        assertThat(publicAccountsAfterUpdate.get(0))
+        assertThat(publicAccountService.findById(createdPublicAccount.getId()))
                 .isEqualToIgnoringGivenFields(expectedUpdatedPublicAccount, "id", "createdAt", "updatedAt");
 
         // Delete
-        publicAccountService.delete(actualPublicAccount.getId());
-        List<PublicAccount> publicAccountsAfterDelete = publicAccountService.findAll();
-        assertEquals(0, publicAccountsAfterDelete.size());
```

Comment:
Improvement can be if you will change it to `assertThat(publicAccountsAfterDelete).hasSize(0);`